compiler/noirc_evaluator/src/ssa/opt/die.rs

```diff
@@ -37,7 +55,10 @@ impl Function {
     /// instructions that reference results from an instruction in another block are evaluated first.
     /// If we did not iterate blocks in this order we could not safely say whether or not the results
     /// of its instructions are needed elsewhere.
-    pub(crate) fn dead_instruction_elimination(&mut self, insert_out_of_bounds_checks: bool) {
+    pub(crate) fn dead_instruction_elimination(
+        &mut self,
+        insert_out_of_bounds_checks: bool,
+    ) -> HashSet<ValueId> {
```

Comment:
Can you add some explanation to the docstrings about the new return value? Are the globals the only reasonable thing to return here?

Reply:
I will add some comments. We do not need anything but the used globals as all other values are going to be per function.

Reply:
Added a comment